src/webview/components/memory-widget.tsx

```diff
@@ -23,6 +23,7 @@ import { OptionsWidget } from './options-widget';
 
 interface MemoryWidgetProps extends MemoryDisplayConfiguration {
     memory?: Memory;
+    initialTitle: string;
```

Comment:
This name is a little confusing - it's really just `title`, since we don't distinguish between `initialTitle` and later titles.

It might be possible to avoid prop drilling here and set up the logic for handling the title in the `OptionsWidget`, since that's the only place it's relevant, but see my next comment for a caveat.

Reply:
I've renamed `initialTitle` into `title` according to your suggestion, which makes a lot of sense especially now when we avoid the need for storing the `title` as a state in the `OptionsWidget`. :+1: 

Regarding the prop drilling... I didn't find a great approach to avoid that though. I personally have a slight preference to keeping the handling of `messenger` events in a central place on both ends (i.e. the `App` for the webview part) to avoid having handlers of messages from the "outside" scattered throughout all react components, but this is just a personal preference.

One way of avoiding prop drilling might be to pull up the header into its own component and import it in the `MemoryWidget` directly, or even removing the middle layer `MemoryWidget` altogether. Grouping prop-values into interfaces may also help to achieve better clarity against an exploding number of props. However, I feel this might be something to attach in a separate PR, wdyt?